std/assembly/internal/typedarray.ts

```diff
@@ -63,8 +63,28 @@ export abstract class TypedArray<T,V> {
 
   // copyWithin(target: i32, start: i32, end: i32 = this.length): this
 
+  fill(value: V, start: i32 = 0, end: i32 = i32.MAX_VALUE): this {
+    var buffer = this.buffer;
+    var byteOffset = this.byteOffset;
+    var len = this.length;
+    start = start < 0 ? max(len + start, 0) : min(start, len);
+    end   = end   < 0 ? max(len + end,   0) : min(end,   len);
+    if (sizeof<T>() == 1) {
+      memory.fill(
+        changetype<usize>(buffer) + start + byteOffset + AB_HEADER_SIZE,
+        <u8>value,
+        <usize>(end - start)
```

Comment:
I think it's necessary to make sure that `end >= start`, otherwise this might overflow and copy like all the memory?

Reply:
e.g. `fill(0xff, 1, 0)` on an `Uint8Array`: start = 1, end = 0, `<usize>(0 - 1) = 0xffffffff`

Reply:
yeah makes sense